### DISCUSSION TITLE
valtio's proxy is  10~30x slower  than @vue/reactivity

### DISCUSSION THREAD

**Discussion by jiangjiu** [pmndrs/valtio#811] [Q&A] [answered]
<img width="1136" alt="image" src="https://github.com/pmndrs/valtio/assets/9526388/28c787c0-fbfd-4177-989e-d0658402485d">

**big Object, proxy 10000 tims;**

why valtio's perf is slower so much? seems like both those two libs just warp a Proxy , same way

demo: https://stackblitz.com/edit/react-9ej1xu?file=src%2FApp.js,src%2FresData.js,src%2Findex.js

**Comment by dai-shi** [comment, marked as answer]
Currently, valtio's `proxy(initialObject)` copies the initialObject _deeply_ to keep initialObject isolated.
So, you don't need to do `Object.assign({}, ...)`.

In v2, we will be changing this behavior. #703

**Reply by jiangjiu** [reply]
sounds great! and when will the stable version V2 be released? about serveral months?

**Reply by dai-shi** [reply]
As written in #703, we are waiting for the RFC to be finalized. Then, v2-alpha, after a while v2-beta, then v2-final, I guess.

I'm not sure if v2 will improve performance or make it worse. We'll see.
